### PR TITLE
Read anemometer data in background

### PR DESCRIPTION
**Pull request by alphaville**
## Patch

- Data acquisition happens in background
- Data are kept in a buffer
- API documentation (not complete - Jamie, I'm leaving the rest to you)

## Basic Example

```python
with Anemometer() as sensor:
    for _i in range(20):
        print(sensor.all_sensor_data)
```

## More advanced example

Here we choose a data processor and a window length

```python
processor = AverageFilter()
with Anemometer(window_length=5, data_processor=processor) as sensor:
    for i in range(20):
        time.sleep(0.1)
        print(sensor.all_sensor_data)
```

### CODE REVIEW

**Comment by alphaville** [comment]
@Minanchi I've invited to review this because you'll be working with the anemometer
@pdavid747 You should review this code because you can reuse it for the GPS
@Runway27 Since you'll be working with GPS too, have a look at this code; it is used to interface the anemometer, but it can be reused for any device that sends data over the serial

File: raspberry/bzzz/sensors/anemometer.py [hunk 1]
```diff
@@ -1,61 +1,137 @@
...
+                 serial_path='/dev/ttyS0',
+                 baud=115200,
+                 window_length=3,
+                 data_processor=MedianFilter()):
+        """
+        Create a new instance of Anemometer
+
+        :param serial_path: serial path; defaults to /dev/ttyS0 on RPi
+        :param baud: baud rate of serial communication; defaults to 115200
+        :param window_length: length of window of measurements; default: 2
```
Comment (jamie-54) [inline raspberry/bzzz/sensors/anemometer.py:52]: Should this say default: 3 instead of 2?

Comment (alphaville) [reply raspberry/bzzz/sensors/anemometer.py]: Good catch! Fixed.

**Review by jamie-54 — APPROVED** [review]

**Comment by alphaville** [comment]
> will we merge this

Yes, this was just a patch to [`feature/87-read_anemometer_data_on_pi`](https://github.com/QUB-ASL/bzzz/tree/feature/87-read_anemometer_data_on_pi); I'm merging now.

**Pull request merged**